src/Amazon.AspNetCore.Identity.Cognito/CognitoUserStore/CognitoUserStore.cs

```diff
@@ -111,6 +111,26 @@ public async Task<IdentityResult> ChangePasswordAsync(TUser user, string current
                 return IdentityResult.Failed(_errorDescribers.PasswordMismatch());
         }
 
+        /// <summary>
+        /// Resets the <paramref name="user"/>'s password to the specified <paramref name="newPassword"/> after
+        /// validating the given password reset <paramref name="token"/>.
+        /// </summary>
+        /// <param name="user">The user whose password should be reset.</param>
+        /// <param name="token">The password reset token to verify.</param>
+        /// <param name="newPassword">The new password to set if reset token verification succeeds.</param>
+        /// <returns>
+        /// The <see cref="Task"/> that represents the asynchronous operation, containing the <see cref="IdentityResult"/>
+        /// of the operation.
+        /// </returns>
+        public async Task<IdentityResult> ChangePasswordWithTokenAsync(TUser user, string token, string newPassword, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            await _pool.ConfirmForgotPassword(user.Username, token, newPassword, cancellationToken).ConfigureAwait(false);
+
+            return IdentityResult.Success;
```

Comment:
Should we be capturing exceptions and returning the failures in the IdentityResult with a status of error? Seems like we are not following the intention of returning an IdentityResult.

Reply:
Pushed a new revision with a POC on IdentityResult handling and exception handling for methods that don't return an IdentityResult object

Reply:
I will roll out the approved pattern as part of DOTNET-3177